src/dependency-manager/spec/utils/spec-validator.ts

```diff
@@ -354,16 +356,132 @@ export const validateOrRejectSpec = (parsed_yml: ParsedYaml, metadata?: Componen
   return component_spec;
 };
 
-export const validateInterpolation = (component_spec: ComponentSpec): void => {
+const extractServiceName = (interpolation: string, type: string): string => {
+  const regex = new RegExp(`${type}\\.([^.[\\]]+)`);
+  const match = interpolation.match(regex);
+  return match ? match[1] : '';
+};
+
+const isDependencyInterpolation = (interpolation_str: string): boolean => {
+  const regex = new RegExp(`^dependencies.[^.]+.(?:services.[^.]+.){0,1}(?:interfaces|ingresses).[^.]+.(?:host|port|protocol|username|password|url|sticky|path|ingress.(?:private|url))$`);
+  return regex.test(interpolation_str);
+};
+
+const isDatabasesInterpolation = (interpolation_str: string): boolean => {
+  const regex = new RegExp(`^databases.[^.]+.(?:connection_string|url)$`);
+  return regex.test(interpolation_str);
+};
+
+const checkInterpolationPath = (queue: any, child: any): boolean => {
+  let key = queue.shift();
+  const keys = Object.keys(child);
+  const star = keys.includes('*');
+  if (star || child[key]) {
+    key = star ? '*' : key;
+    if (queue.length > 0 && typeof child[key] === 'object') {
+      return checkInterpolationPath(queue, child[key]);
+    } else {
+      return true;
+    }
+  } else {
+    return keys.includes(key);
+  }
+};
+
+const findInterpolationErrors = (errors: ValidationError[], component_spec: ComponentSpec, graph: Readonly<DependencyGraphMutable>) => {
```

Comment:
If we have the mock dependencies why can't we just turn the validation on and just let the normal validation logic handle everything?

Reply:
Like turn on the `validate` in the `getGraph` function? What do you mean by `normal validation logic` here?

Reply:
Currently, there are 2 issues with setting `interpolate` and `validate` to `true`.
```
1. DEP_ADDR: ${{ dependencies.hello-world.services.hello.interfaces.main.url }}
     |                                      ﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋﹋ Invalid interpolation ref: ${{ dependencies.hello-world.services.hello.interfaces.main.url }} - Did you mean ${{ dependencies.hello-world.services.*.interfaces.*.url }}?
2. param_unset:
     |   ﹋﹋﹋﹋﹋﹋ Required secret 'param_unset' was not provided
```
I'm working on finding solution for these cases.

Reply:
Yeah we need to make it so the validation check allows the * when mock_dependnecies is enabled and then it should work.